src/webviews/mongoClusters/collectionView/components/toolbar/ToolbarViewNavigation.tsx

```diff
@@ -6,37 +6,99 @@
 import { Dropdown, Label, Option, Toolbar, ToolbarButton, Tooltip } from '@fluentui/react-components';
 import { ArrowLeftFilled, ArrowPreviousFilled, ArrowRightFilled } from '@fluentui/react-icons';
 import { useContext } from 'react';
+import { useTrpcClient } from '../../../../api/webview-client/useTrpcClient';
 import { CollectionViewContext } from '../../collectionViewContext';
 import { ToolbarDividerTransparent } from './ToolbarDividerTransparent';
 
 export const ToolbarViewNavigation = (): JSX.Element => {
+    /**
+     * Use the `useTrpcClient` hook to get the tRPC client and an event target
+     * for handling notifications from the extension.
+     */
+    const { trpcClient /** , vscodeEventTarget */ } = useTrpcClient();
+
     const [currentContext, setCurrentContext] = useContext(CollectionViewContext);
 
     function goToNextPage() {
+        const newPage = currentContext.currrentQueryDefinition.pageNumber + 1;
+
         setCurrentContext({
             ...currentContext,
             currrentQueryDefinition: {
                 ...currentContext.currrentQueryDefinition,
-                pageNumber: currentContext.currrentQueryDefinition.pageNumber + 1,
+                pageNumber: newPage,
             },
         });
+
+        trpcClient.common.reportEvent
+            .mutate({
+                eventName: 'pagination',
+                properties: {
+                    source: 'next-page',
+                    ui: 'button',
+                    view: currentContext.currentView,
+                },
+                measurements: {
+                    page: newPage,
+                    pageSize: currentContext.currrentQueryDefinition.pageSize,
+                },
+            })
+            .catch((error) => {
+                console.error('Failed to report pagination event:', error);
+            });
     }
 
     function goToPreviousPage() {
+        const newPage = Math.max(1, currentContext.currrentQueryDefinition.pageNumber - 1);
+
         setCurrentContext({
             ...currentContext,
             currrentQueryDefinition: {
                 ...currentContext.currrentQueryDefinition,
-                pageNumber: Math.max(1, currentContext.currrentQueryDefinition.pageNumber - 1),
+                pageNumber: newPage,
             },
         });
+
+        trpcClient.common.reportEvent
+            .mutate({
+                eventName: 'pagination',
+                properties: {
+                    source: 'prev-page',
+                    ui: 'button',
+                    view: currentContext.currentView,
+                },
+                measurements: {
+                    page: newPage,
+                    pageSize: currentContext.currrentQueryDefinition.pageSize,
+                },
+            })
+            .catch((error) => {
+                console.error('Failed to report pagination event:', error);
```

Comment:
If this is error then either it should show message to user or send to telemetry. If this is debug info, please use console.debug or log.